crates/autopilot/src/run_loop.rs

```diff
@@ -357,6 +357,38 @@ impl RunLoop {
         tokio::spawn(settle_fut);
     }
 
+    fn compute_reference_scores(solutions: &[competition::Participant]) -> Vec<ReferenceScore> {
+        let Some(total_score) = solutions
+            .iter()
+            .filter_map(|participant| {
+                participant
+                    .is_winner()
+                    .then_some(participant.solution().score().get().0)
+            })
+            .reduce(U256::saturating_add)
+        else {
+            return Default::default();
+        };
+
+        solutions
+            .iter()
+            .filter(|participant| participant.is_winner())
+            .group_by(|participant| participant.driver().submission_address)
+            .into_iter()
+            .map(|(solver, solutions)| {
+                let score = solutions
+                    .map(|participant| participant.solution().score().get().0)
+                    .reduce(|a, b| a.checked_add(b).unwrap_or_default())
+                    .unwrap_or_default();
```

Comment:
What's the reason to use `saturating_add` for the total score and `checked_add` here?